### PR TITLE
[Refactor] Unify the shared pass prefix between vm and graph

### PR DESCRIPTION
**Pull request by YuchenJin**
This PR unifies the common optimization pass prefix shared between vm and graph executor into a single API.

### CODE REVIEW

**Comment by YuchenJin** [comment]
@jroesch could you take a look? 

It would be great if we can move [RunDeviceAnnotationPass](https://github.com/apache/tvm/blob/c989e4afee1c45b755371e856d577c08dff410a1/src/relay/backend/build_module.cc#L469) to Relay pass infra and the `relay::transform` namespace, the code would be cleaner and more passes can be packed into this prefix.

File: src/relay/backend/utils.h [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include <tvm/driver/driver_api.h>
 #include <tvm/relay/expr.h>
 #include <tvm/relay/expr_functor.h>
+#include <tvm/relay/qnn/transform.h>
```
Comment (jcf94) [inline src/relay/backend/utils.h:31]: Move this include to utils.cc?

**Review by jcf94 — CHANGES_REQUESTED** [review]
Overall looks fine to me.

File: src/relay/backend/utils.h [hunk 2]
```diff
@@ -410,6 +416,15 @@ inline bool IsCompileEngineCacheDisabled() {
...
 }
 
+/*!
+ * \brief Get a shared optimization pass prefix between vm and graph executor.
+ *
+ * \param targets The device type to `Target` mapping.
+ * \param is_vm A boolean indicating if the passes are used for vm or graph executor.
+ * \return An array of passes.
+ */
+Array<Pass> GetPrefixOpts(const Map<tvm::Integer, tvm::Target>& targets, bool is_vm);
```
Comment (jcf94) [inline src/relay/backend/utils.h:426]: Better to explain more clearly about what these passes will do, or come up with a better function name.
But I don't have suggestions from my point of view ...

File: src/relay/backend/utils.h (hunk 1 above)
Comment (YuchenJin) [reply src/relay/backend/utils.h]: Good catch! Moved to utils.cc.

File: src/relay/backend/utils.h (hunk 2 above)
Comment (YuchenJin) [reply src/relay/backend/utils.h]: Thanks for review! I renamed it to `GetPassPrefix` and added more comments. Hopefully it gets clearer. :)

**Review by jcf94 — APPROVED** [review]
Fine to me.

**Pull request merged**